pkg/trie/triedb/node.go

```diff
@@ -108,9 +109,9 @@ func (vr newValueRef[H]) equal(other nodeValue) bool {
 }
 
 func NewValue[H hash.Hash](data []byte, threshold int) nodeValue {
-	if len(data) >= threshold {
+	if len(data) > threshold {
```

Comment:
good catch

Reply:
Yea this is called repeatedly via `TrieDB` methods.  I actually uncovered this problem through the tests.